### PR TITLE
Livetime scaling

### PR DESCRIPTION
**Pull request by kdund**
This aims to allow runtime changing of livetime in data generation, get_expectation and likelihood evaluation. 

To test implementation: 

```
from alea.models.blueice_extended_model import BlueiceExtendedModel

model = BlueiceExtendedModel.from_config("/Users/kdund/Software/xenon_software/alea/alea/examples/configs/unbinned_wimp_statistical_model.yaml")

data = model.generate_data(er_rate_multiplier=1, wimp_rate_multiplier=1)
model.data = data

#get expectation value check: 
print(model.get_expectation_values(livetime_sr0=100, livetime_sr1=0))
print(model.get_expectation_values(livetime_sr0=1, livetime_sr1=0))
#1st is 100 times second

#likelihood evaluation check: 
ll_ratio = model.ll(livetime_sr0=1, er_rate_multiplier=1, wimp_rate_multiplier=0, livetime_sr1 = 0) / model.ll(livetime_sr0=10, er_rate_multiplier=0.1, wimp_rate_multiplier=0, livetime_sr1 = 0) #check if livetime/rate is degenerate as expected. 
print(ll_ratio, "approx equal 1")

#data generation check: 
print(len(model.generate_data(er_rate_multiplier=1, wimp_rate_multiplier=1, livetime_sr0=10)['sr0'])) #around 
 
```

### CODE REVIEW

File: alea/models/blueice_extended_model.py [hunk 1]
```diff
@@ -130,11 +133,14 @@ def get_expectation_values(self, **kwargs) -> dict:
         self_copy.data = self_copy.generate_data()
 
         # ancillary likelihood does not contribute
-        for ll_term, parameter_names in zip(
+        for ll_term, parameter_names, livetime_parameter in zip(
```
[the remark this answers is not in the document]
Comment (kdund) [reply alea/models/blueice_extended_model.py:138]: yes. what do do about it?

**Comment by kdund** [comment]
Confidence interval call currently does not propagate lifetimes well-- had to set values via parameters.livetime_sr0.nominal_value = X

File: alea/models/blueice_extended_model.py [hunk 2]
```diff
@@ -272,8 +281,9 @@ def store_data(self, file_name, data_list, data_name_list=None, metadata=None):
 
     def _generate_science_data(self, **generate_values) -> dict:
         """Generate the science data for all likelihood terms except the ancillary likelihood."""
+        livetime_days = [generate_values.get(ln,None) for ln in self.livetime_parameter_names]
         science_data = [
-            gen.simulate(**generate_values) for gen in self.data_generators]
+            gen.simulate(livetime_days = lt, **generate_values) for gen, lt in zip(self.data_generators, livetime_days)]
```
Comment (dachengx) [inline alea/models/blueice_extended_model.py:286]: Maybe change `(ln,None)` to `(ln, None)`, and `livetime_days = lt,` to `livetime_days=lt,`.

File: alea/models/blueice_extended_model.py [hunk 3]
```diff
@@ -239,7 +247,8 @@ def _build_data_generators(self) -> list:
             BlueiceDataGenerator(ll_term) for ll_term in self._likelihood.likelihood_list[:-1]]
 
     def _ll(self, **generate_values) -> float:
-        return self._likelihood(**generate_values)
+        livetime_days = [generate_values.get(ln, None) for ln in self.livetime_parameter_names]
+        return self._likelihood(livetime_days = livetime_days, **generate_values)
```
Comment (dachengx) [inline alea/models/blueice_extended_model.py:251]: Maybe change `livetime_days = livetime_days` to `livetime_days=livetime_days`.

File: alea/models/blueice_extended_model.py [hunk 4]
```diff
@@ -48,6 +50,9 @@ def __init__(self, parameter_definition: dict, likelihood_config: dict, **kwargs
         super().__init__(parameter_definition=parameter_definition, **kwargs)
         self._likelihood = self._build_ll_from_config(likelihood_config)
         self.likelihood_names = [t["name"] for t in likelihood_config["likelihood_terms"]]
+        self.livetime_parameter_names = [t.get("livetime_parameter", None) for t in
+                                         likelihood_config["likelihood_terms"]]
+        self.livetime_parameter_names += [None]  # ancillary likelihood
         self.likelihood_names.append("ancillary_likelihood")
```
Comment (dachengx) [inline alea/models/blueice_extended_model.py:56]: Maybe move this line to below `self.likelihood_names = [t["name"] for t in likelihood_config["likelihood_terms"]]`.

Comment (kdund) [reply alea/models/blueice_extended_model.py:56]: isn't it below?

Comment (dachengx) [reply alea/models/blueice_extended_model.py:56]:
```
        self.likelihood_names = [t["name"] for t in likelihood_config["likelihood_terms"]]
        self.likelihood_names.append("ancillary_likelihood")
        self.livetime_parameter_names = [t.get("livetime_parameter", None) for t in
                                         likelihood_config["likelihood_terms"]]
        self.livetime_parameter_names += [None]  # ancillary likelihood
```

Comment (kdund) [reply alea/models/blueice_extended_model.py:56]: ah sure. Wish it showed the line you commented on.

**Pull request merged**